compiler/passes/normalize.cpp

```diff
@@ -405,8 +405,23 @@ static bool isInWhereBlock(FnSymbol* fn, Expr* expr) {
   return fn->where == expr;
 }
 
-static void moveInterfaceConstraints() {
+// Ensure we can invoke icon->ifcSymbol() from now on.
+static void checkInterfaceConstraint(IfcConstraint* icon) {
+  if (SymExpr* ifcSE = toSymExpr(icon->interfaceExpr)) {
+    if (! isInterfaceSymbol(ifcSE->symbol())) {
+      USR_FATAL_CONT(ifcSE, "'%s' is not an interface", ifcSE->symbol()->name);
+      USR_PRINT(icon, "an 'implements' keyword must be followed"
+                      " by an interface name");
+    }
```

Comment:
It might be helpful to additionally point to where the symbol we found was defined